Index: calico-dnsmasq/man/dnsmasq.8
===================================================================
--- calico-dnsmasq.orig/man/dnsmasq.8	2014-12-02 17:09:58.655208182 +0000
+++ calico-dnsmasq/man/dnsmasq.8	2014-12-02 17:18:30.019229751 +0000
@@ -1080,6 +1080,7 @@
 as if they had arrived at <interface>. This option is necessary when
 using "old style" bridging on BSD platforms, since
 packets arrive at tap interfaces which don't have an IP address.
+A trailing '*' wildcard can be used in each <alias>.
 .TP
 .B \-s, --domain=<domain>[,<address range>[,local]]
 Specifies DNS domains for the DHCP server. Domains may be be given 
Index: calico-dnsmasq/man/fr/dnsmasq.8
===================================================================
--- calico-dnsmasq.orig/man/fr/dnsmasq.8	2014-12-02 17:09:58.655208182 +0000
+++ calico-dnsmasq/man/fr/dnsmasq.8	2014-12-02 17:18:30.019229751 +0000
@@ -1220,7 +1220,7 @@
 comme si elles arrivaient de l'interface <interface>. Cette option est
 nécessaire lors de l'utilisation de pont ethernet "ancien mode" sur plate-forme
 BSD, puisque dans ce cas les paquets arrivent sur des interfaces "tap" n'ont
-pas d'adresse IP.
+pas d'adresse IP.  Chaque <alias> peut finir avec un simple '*' joker.
 .TP
 .B \-s, --domain=<domaine>[,<gamme d'adresses>[,local]]
 Spécifie le domaine du serveur DHCP. Le domaine peut être donné de manière
Index: calico-dnsmasq/src/dnsmasq.h
===================================================================
--- calico-dnsmasq.orig/src/dnsmasq.h	2014-12-02 17:09:58.659208182 +0000
+++ calico-dnsmasq/src/dnsmasq.h	2014-12-02 17:18:30.019229751 +0000
@@ -796,6 +796,7 @@
 char *print_mac(char *buff, unsigned char *mac, int len);
 void bump_maxfd(int fd, int *max);
 int read_write(int fd, unsigned char *packet, int size, int rw);
+int wildcard_matchn(const char* wildcard, const char* match, int num);
 
 /* log.c */
 void die(char *message, char *arg1, int exit_code);
Index: calico-dnsmasq/src/network.c
===================================================================
--- calico-dnsmasq.orig/src/network.c	2014-12-02 17:09:58.659208182 +0000
+++ calico-dnsmasq/src/network.c	2014-12-02 17:18:30.019229751 +0000
@@ -162,7 +162,7 @@
       for (bridge = daemon->bridges; bridge; bridge = bridge->next)
 	{
 	  for (alias = bridge->alias; alias; alias = alias->next)
-	    if (strncmp(name, alias->iface, IF_NAMESIZE) == 0)
+	    if (wildcard_matchn(alias->iface, name, IF_NAMESIZE))
 	      {
 		int newindex;
 		
Index: calico-dnsmasq/src/util.c
===================================================================
--- calico-dnsmasq.orig/src/util.c	2014-12-02 17:09:58.659208182 +0000
+++ calico-dnsmasq/src/util.c	2014-12-02 17:18:30.019229751 +0000
@@ -522,3 +522,21 @@
   return 1;
 }
 
+/* The same but comparing a maximum of NUM characters, like strncmp.  */
+int wildcard_matchn(const char* wildcard, const char* match, int num)
+{
+  while (*wildcard && *match && num)
+    {
+      if (*wildcard == '*')
+        return 1;
+
+      if (*wildcard != *match)
+        return 0; 
+
+      ++wildcard;
+      ++match;
+      --num;
+    }
+
+  return (!num) || (*wildcard == *match);
+}
